packages/common/async/src/task-scheduling.test.ts

```diff
@@ -44,4 +44,11 @@ describe('task-scheduling', () => {
       expect(called).to.be.false;
     });
   });
+
+  test('isThreadSaturated', async () => {
+    // just a sanity check
+    if (isThreadSaturated()) {
```

Comment:
I'd add another method that'll perform both operations, like if `yieldIfThreadSaturated`, I think that's going to be the most common usecase